File: spring-mvc/src/main/java/software/darkmatter/school/blog/domain/comment/business/CommentService.java
package software.darkmatter.school.blog.domain.comment.business;

import org.springframework.data.domain.Pageable;
import software.darkmatter.school.blog.api.dto.CommentCreateDto;
import software.darkmatter.school.blog.api.dto.CommentUpdateDto;
import software.darkmatter.school.blog.domain.comment.data.Comment;

import java.util.List;

public interface CommentService {

    List<Comment> getListByPostId(Long postId, Pageable pageable);

    Comment getById(Long id);

    Comment getByPostIdAndId(Long postId, Long id);

    Comment create(Long postId, CommentCreateDto createDto);

    Comment update(Long id, CommentUpdateDto updateDto);

    void delete(Long id);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/comment/business/CommentService.kt
package software.darkmatter.school.blog.domain.comment.business

import org.springframework.data.domain.Pageable
import software.darkmatter.school.blog.api.dto.CommentCreateDto
import software.darkmatter.school.blog.api.dto.CommentUpdateDto
import software.darkmatter.school.blog.domain.comment.data.Comment

interface CommentService {
    suspend fun getListByPostId(postId: Long, pageable: Pageable): List<Comment>
    suspend fun getById(id: Long): Comment
    suspend fun getByPostIdAndId(postId: Long, id: Long): Comment
    suspend fun create(postId: Long, createDto: CommentCreateDto): Comment
    suspend fun update(id: Long, updateDto: CommentUpdateDto): Comment
    suspend fun delete(id: Long)
}
